src/models/geograph.py

```diff
@@ -14,25 +27,92 @@ class GeoGraph:
+        attributes: Optional[List[str]] = None,
+        graph_load_path: Optional[Union[str, os.PathLike]] = None,
+        graph_save_path: Optional[Union[str, os.PathLike]] = None,
+        raster_array: Optional[np.ndarray] = None,
+        raster_load_path: Optional[Union[str, os.PathLike]] = None,
+        raster_save_path: Optional[Union[str, os.PathLike]] = None,
+        vector_path: Optional[Union[str, os.PathLike]] = None,
+        tolerance: float = 0.0,
+        mask: Optional[np.ndarray] = None,
+        transform: affine.Affine = affine.identity,
+        crs: Optional[str] = None,
     ) -> None:
         """
         Class for the fragmentation graph.
 
-        The `graph_path` argument allows for loading a saved networkx graph.
+        This class can load a pickled networkx graph directly, or create the
+        graph from vector data, raster data, or a dataframe containing polygons.
 
         Args:
-            dataframe (Optional[gpd.GeoDataFrame], optional): A geopandas
-            dataframe object. Defaults to None.
-            graph_path (Optional[str], optional): A path to a pickled networkx
-            graph. Defaults to None.
+            dataframe (gpd.GeoDataFrame, optional): A geopandas dataframe
+            object containing a `geometry` column with polygon data. To load
+            polygon class labels as node attributes, include these in a column.
+            Defaults to None.
+            attributes (Optional[List[str]], optional): columns of dataframe
+            that are added to nodes of graph as attributes. Defaults to None,
+            which will cause all attributes to be added.
+            graph_load_path (str or pathlib.Path, optional): A path to a pickle
+            file to load from, can be `.gz` or `.bz2`. Defaults to None.
+            graph_save_path (str or pathlib.Path, optional): A path to a pickle
+            file to save the graph to, can be `.gz` or `.bz2`. Defaults to None,
+            which will save the graph in the project root in the `data`
+            folder.
+            raster_array (np.ndarray, optional): 2D numpy array with the raster
+            data that can be passed as an alternative to `raster_path`.
+            vector_path (str or pathlib.Path, optional): A path to a file of
+            vector data, either in GPKG or Shapefile format. Defaults to None.
+            raster_load_path (str or pathlib.Path, optional): A path to a file
+            of raster data in GeoTiff format. Defaults to None.
+            raster_save_path (str or pathlib.Path, optional): A path to a file
+            to save the polygonised raster data in. A path to a GPKG file is
+            recommended, but Shapefiles also work. Defaults to None, which saves
+            to the project root in the `data` folder in GPKG format.
+            tolerance (float, optional): Adds edges between neighbours that are
+            at most `tolerance` metres apart. Defaults to 0.
```

Comment:
```suggestion
            at most `tolerance` units apart. Defaults to 0.
```
(since ppl might load graphs with other base units)

Reply:
Done.